### PR TITLE
Better SHA extraction from GitHub webhook payloads

### LINKED ISSUE #2697: Wrong commit SHA in `staging`
**What happened**: Today, I head over the [`staging`](https://dev.telescope.cdot.systems) deployment, and I clicked on the version tag (as of right now, it is version 2.5.0). After clicking it redirected to a commit that does not exist (the SHA in the link does not reference an existing commit).

**What should have happened**: A page that shows an existing commit.

**How to reproduce it (as precise as possible)**: Go to [`staging`](https://dev.telescope.cdot.systems) and click on the version tag.

**Anything else we need to know?**: This might be related to #2650 or not. I am not really sure.

**Environment**:

- OS: Windows 10
- Browser: FireFox v. 95.0.2

### PR DESCRIPTION
**Pull request by humphd**
Fixes #2697, possibly #2650? (cc @tcvan0707).

### CODE REVIEW

File: tools/autodeployment/webhook.js [hunk 1]
```diff
@@ -48,6 +48,24 @@ function requestFilter(repo, buildType, action, ref, mainBranch) {
...
+ * @param {Object} githubData webhook payload from GitHub, see https://github.com/Seneca-CDOT/telescope/settings/hooks
+ */
+function getSha(eventType, githubData) {
+  if (eventType === 'push') {
+    // "after": "adf299da6e6e9c3f20208ee6ab61d205a9f70eab",
+    return githubData.after;
+  }
+  if (eventType === 'release') {
+    // "target_commitish": "1a4594cfc443d307679576d9fdfa402f530a570c",
+    return githubData.release.target_commitish;
```
Comment (dbelokon) [inline tools/autodeployment/webhook.js:63]: How do we know that `target_commitish` is the SHA? I checked the GitHub API docs and the example shown shows the name of the `master` branch 😕

**Review by dbelokon — COMMENTED** [review]
I'm wondering, what was the problem for #2697 to occur? I read @manekenpix comment, but I am not sure how we can prevent that 😕

File: tools/autodeployment/webhook.js (hunk 1 above)
Comment (manekenpix) [reply tools/autodeployment/webhook.js:63]: @dbelokon https://github.com/Seneca-CDOT/telescope/issues/2650#issuecomment-1018045662

Comment (dbelokon) [reply tools/autodeployment/webhook.js:63]: Oh, I didn't know about that problem with the documentation. Thank you @manekenpix!

**Comment by manekenpix** [comment]
@humphd can you rebase it so I can test it on staging?

**Comment by humphd** [comment]
Rebased. Try now?

**Review by TDDR — COMMENTED** [review]
I tested this on staging. The link now send you to what seems to be the latest commit merged into master. Is this the behavior wanted, or did you want it linking to the commit that brought it to that version?

**Review by TDDR — COMMENTED** [review]
I tested this on staging. The link now send you to what seems to be the latest commit merged into master. Is this the behavior wanted, or did you want it linking to the commit that brought it to that version?

**Comment by tcvan0707** [comment]
> I tested this on staging. The link now send you to what seems to be the latest commit merged into master. Is this the behavior wanted, or did you want it linking to the commit that brought it to that version?

It seems to be the expected behavior. I also test it, if I remember correctly, before `staging` crashed, it leaded us to the recent commit merged to `master` branch.

File: tools/autodeployment/builds.js [hunk 2]
```diff
@@ -4,9 +4,10 @@ const mergeStream = require('merge-stream');
 const { ReReadable } = require('rereadable-stream');
 
 class Build {
-  constructor(type, githubData) {
+  constructor(type, githubData, sha) {
```
Comment (manekenpix) [inline tools/autodeployment/builds.js:7]: Tested on staging and production.
For this to happen you need to add the sha as a parameter in lines 96 and 100. I think after that this is good to go

**Review by manekenpix — CHANGES_REQUESTED** [review]

File: tools/autodeployment/builds.js (hunk 2 above)
Comment (manekenpix) [reply tools/autodeployment/builds.js:7]: I manually added the changes mentioned in the previous comment to staging and production and they're displaying the correct sha for both.

Comment (humphd) [reply tools/autodeployment/builds.js:7]: Awesome, nice catch. Sorry I missed this.

I've added both, rebased, and pushed again.

**Review by manekenpix — APPROVED** [review]
:+1:

**Review by TDDR — APPROVED** [review]
Looks good to me.

**Pull request merged**